src/decorator.ts

```diff
@@ -21,19 +21,22 @@ import { untilDestroyed } from './take-until-destroy';
  *
  * Do not forget to implement {@link OnDestroy} life-cycle hook.
  */
-export function WithUntilDestroyed(
-  destroyMethodName?: string,
-): PropertyDecorator {
-  return (target, propKey) => {
-    let val: Observable<any>;
+export function WithUntilDestroyed(destroyMethodName?: string): PropertyDecorator {
+  return function(target, propKey) {
+    const valKey = `__WithUntilDestroyed:${String(propKey)}__`;
 
     function getter() {
-      return val;
+      return this[valKey];
     }
 
     function setter(newVal) {
```

Comment:
Forget about it. I'm removing this from the library; it's not worth it. Thanks anyway.

Reply:
Sure.
I will publish forked version with this feature. Thanks